frontend/compute-system/src/components/ComputationCockpit/ComputationCockpit.tsx

```diff
@@ -1,7 +1,131 @@
-import React from "react";
+import React, { useCallback, useEffect, useState } from "react";
+import {
+  Badge,
+  Button,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  useMediaQuery,
+} from "@material-ui/core";
+import apiCall from "../../api/apiCall";
+import { COCKPIT_URL } from "../../api/urls";
+import RequestType from "../../api/requestType";
+import { TaskData } from "./taskData";
+import { mockTask } from "../../mocks/ComputationCockpit/mockTask";
+import { StartTask } from "./TaskActions/StartTask";
+import { TerminateTask } from "./TaskActions/TerminateTask";
+import { ArchiveTask } from "./TaskActions/ArchiveTask";
+import { useHistory } from "react-router-dom";
+import { routes } from "../../const/routes";
+import { MockWarning } from "../common/MockWarning";
+import {
+  formatTaskClusterAllocation,
+  formatTaskCredits,
+  formatTaskRuntime,
+} from "./taskDataFormat";
 
 const ComputationCockpit: React.FC = () => {
-  return <p>WIP</p>;
+  const history = useHistory();
+  const matches = useMediaQuery("(min-width:800px)");
+
+  const [tasks, setTasks] = useState<TaskData[]>();
+  const [downloaded, setDownloaded] = useState<boolean>(false);
+
+  const fetchTasks = useCallback(async () => {
+    const response = await apiCall<TaskData[]>(COCKPIT_URL, RequestType.GET);
+    if (response.isError) {
+      return;
+    }
+    try {
+      //TODO: remove this closure when backend no longer returns invalid trash
+      let tasks = response.content as TaskData[];
+      tasks.forEach((task) => {
+        task.startTime &&
+          (task.startTime = new Date(task.startTime).toLocaleString());
+        task.endTime &&
+          (task.endTime = new Date(task.endTime).toLocaleString());
+      });
+      setTasks(tasks);
+      setDownloaded(true);
+    } catch (e) {
+      setTasks(Array(10).fill(mockTask));
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchTasks();
+  }, [fetchTasks]);
+
+  const handleTaskClick = (taskId: number) =>
+    history.push(`${routes.computationCockpit}/task/${taskId}`);
+
+  return (
+    <TableContainer component={Paper}>
+      {!downloaded && <MockWarning />}
+      <Table aria-label="simple table">
```

Comment:
I think, tasks could be separated into multiple tables by IdApp stored in database or just add column for IdApp

Reply:
Well, that could be easier to figure out if backend provided any model of the data served by API. Meanwhile, server returns "69" and I have to rely on the screenshot of Swagger in the project repo, because the actual docs show only endpoint for listing (also without data model). Until backend team gets theit sh** together, I don't think I'll make any changes because there might me more to correct. I've done my best.

Sorry for beibg a little harsh. Don't take it personal.